datamonster_api/lib/company.py

```diff
@@ -2,38 +2,60 @@
 
 
 class Company(BaseClass):
-    """Class for a company"""
+    """Company object which represents a company in DataMonster
+
+    :param params: (dict)
+    :param dm: DataMonster object
+
+    Attributes include:
+      ticker: (str) company ticker
+      name: (str) company name
+      quarters: (list) list of company quarter dates
+    """
 
     _details = None
 
-    def __init__(self, id_, ticker, name, uri, dm):
-        self.id = id_
-        self.ticker = ticker
-        self.name = name
-        self.uri = uri
-        self.dm = dm
+    def __init__(self, params, dm):
+        self._dm = dm
+        self._params = params
+        self._id = params["id"]
+        self._uri = params["uri"]
+        self.ticker = params["ticker"]
+        self.name = params["name"]
+        self.quarters = params.get("quarters", [])
 
     def __repr__(self):
         ticker_str = "[{}] ".format(self.ticker) if self.ticker else ""
         return "<{}: {}{}>".format(self.__class__.__name__, ticker_str, self.name)
 
     def get_details(self):
         """Get details (metadata) for this company
+
+        :return: (dict) details
         """
-        return self.dm.get_company_details(self.id)
+        return self._dm.get_company_details(self._id)
 
     @property
     def datasources(self):
```

Comment:
It might be worth bringing it memoized_property for this just to clean it up and make it more reusable for future memoized properties